client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-plugin-section.tsx

```diff
@@ -79,6 +86,41 @@ export const ThankYouPluginSection = ( { plugin }: { plugin: any } ) => {
 		plugin?.setup_url && siteAdminUrl ? siteAdminUrl + plugin.setup_url : null;
 	const setupURL = plugin?.action_links?.Settings || fallbackSetupUrl || managePluginsUrl;
 	const documentationURL = plugin?.documentation_url;
+	const purchases = useSelector( ( state ) => getUserPurchases( state ) );
+	const isLoadingPurchases = useSelector(
+		( state ) => isFetchingUserPurchases( state ) || ! hasLoadedUserPurchasesFromServer( state )
+	);
+	useQueryUserPurchases();
```

Comment:
I believe we can do a more specific request by using `<QuerySitePurchases siteId={ selectedSite?.ID } />`

Reply:
Updated!